framework-api-core/src/main/java/uk/gov/justice/services/core/enveloper/Enveloper.java

```diff
@@ -24,4 +27,43 @@
      * @return a function that wraps objects into an envelope.
      */
     Function<Object, JsonEnvelope> withMetadataFrom(final JsonEnvelope envelope, final String name);
+
```

Comment:
The above two withMetadataFrom methods should be marked as deprecated, including Javadoc to explain what method to use instead.